### PR TITLE
documents: import other editions

### PR DESCRIPTION
**Pull request by sebastiendeleze**
* Extracts and imports other editions from field 775.
* Displays other editions in detail view.

Co-Authored-by: Sébastien Délèze <sebastien.deleze@rero.ch>

### CODE REVIEW

File: sonar/modules/documents/dojson/contrib/marc21tojson/model.py [hunk 1]
```diff
@@ -848,3 +848,22 @@ def marc21_to_files(self, key, value):
...
+
+@marc21tojson.over('otherEdition', '^775..')
+@utils.for_each_value
+@utils.ignore_value
+def marc21_to_other_edition(self, key, value):
+    """Get other edition."""
+    electronic_locator = value.get('o')
+    public_note = value.get('g')
+
+    if not electronic_locator or not public_note:
```
Comment (jma) [inline sonar/modules/documents/dojson/contrib/marc21tojson/model.py]: Does ignore_value manage with this? If yes this is useless.

File: sonar/modules/documents/jsonschemas/documents/document-v1.0.0_src.json [hunk 2]
```diff
@@ -1663,6 +1663,40 @@
...
+      "type": "array",
+      "minItems": 1,
+      "items": {
+        "type": "object",
+        "required": ["document", "publicNote"],
+        "additionalProperties": false,
+        "properties": {
+          "document": {
+            "title": "Document",
+            "description": "Document object.",
```
Comment (jma) [inline sonar/modules/documents/jsonschemas/documents/document-v1.0.0_src.json]: bad description

File: sonar/modules/documents/jsonschemas/documents/document-v1.0.0_src.json [hunk 3]
```diff
@@ -1663,6 +1663,40 @@
...
+            "type": "object",
+            "required": ["electronicLocator"],
+            "additionalProperties": false,
+            "properties": {
+              "electronicLocator": {
+                "title": "Electronic locator",
+                "description": "Electronic locator.",
```
Comment (jma) [inline sonar/modules/documents/jsonschemas/documents/document-v1.0.0_src.json]: useless

File: sonar/modules/documents/jsonschemas/documents/document-v1.0.0_src.json [hunk 4]
```diff
@@ -1663,6 +1663,40 @@
...
+                "type": "string",
+                "minLength": 1
+              }
+            }
+          },
+          "publicNote": {
+            "title": "Public note",
+            "description": "Public note.",
```
Comment (jma) [inline sonar/modules/documents/jsonschemas/documents/document-v1.0.0_src.json]: useless

File: sonar/modules/documents/mappings/v6/documents/document-v1.0.0.json [hunk 5]
```diff
@@ -362,6 +362,9 @@
         },
         "_updated": {
           "type": "date"
+        },
+        "otherEdition": {
+          "type": "object"
```
Comment (jma) [inline sonar/modules/documents/mappings/v6/documents/document-v1.0.0.json]: Seems be a minimalist approach. Please use properties.

File: sonar/modules/documents/dojson/contrib/marc21tojson/model.py (hunk 1 above)
Comment (sebastiendeleze) [reply sonar/modules/documents/dojson/contrib/marc21tojson/model.py]: ignore_value checks if returned value is None and not the value for all the properties. In addition, in this case, I want to force to have both properties defined in order to store otherEdition.

**Pull request merged**